### ISSUE TITLE
Using Carbon's FrameLayout as a container for fragments results in an ArrayIndexOutOfBoundsException (part 2)

### ISSUE THREAD

**Issue by AlexMomotov** [ZieIony/Carbon#170]
See here for the original issue: https://github.com/ZieIony/Carbon/issues/62.

<b>The error is still there in 0.11.0 version. :( </b>

**Comment by ZieIony** [comment]
Interesting, I thought this bug was fixed long time ago. Could you please add a stacktrace and describe your problem precisely? Is it the same situation as in #62?

**Comment by AlexMomotov** [comment]
I use carbon.widget.FrameLayout to display a group of fragments in it. I change them using simple transaction. And it works like a charm. But when I open the launcher (i.e. desktop screen of any Android device), turn off device screen by pressing power button, and when return to my app via recents app I am getting this error. I thought it happens during Activity recreate, but it works fine when I change screen orientation. So. It seems like it is a bug in carbon.widget.FrameLayout itself. Changing it to regular Android FrameLayout fix it.

Anyway. Your project (whole library) is fantastic! It is really hard to build Material Design app with only Google's support libraries. But your project makes it easy. Thank you a lot!

Alex. P.S. Sorry for my English. Sometimes it's bad.

```
/* Make fragment replacement */
    private void displayFragment(Fragment fragment) {
        if (fragment != null) {            
            FragmentManager fm = getSupportFragmentManager();
            fm.beginTransaction()
                    .replace(R.id.navigation_frame, fragment)
                    .commitAllowingStateLoss();
        }
    }
```

```
01-11 19:02:02.815 4558-4558/ru.muza E/AndroidRuntime: FATAL EXCEPTION: main
                                                       Process: ru.muza, PID: 4558
                                                       java.lang.ArrayIndexOutOfBoundsException: length=12; index=-1
                                                           at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2019)
                                                           at carbon.widget.FrameLayout.dispatchTouchEvent(FrameLayout.java:279)
                                                           at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2405)
                                                           at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2049)
                                                           at carbon.widget.LinearLayout.dispatchTouchEvent(LinearLayout.java:278)
                                                           at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2405)
                                                           at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2049)
                                                           at carbon.widget.DrawerLayout.dispatchTouchEvent(DrawerLayout.java:274)
                                                           at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2405)
                                                           at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2049)
                                                           at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2405)
                                                           at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2049)
                                                           at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2405)
                                                           at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2049)
                                                           at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2405)
                                                           at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2049)
                                                           at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2405)
                                                           at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2049)
                                                           at com.android.internal.policy.impl.PhoneWindow$DecorView.superDispatchTouchEvent(PhoneWindow.java:2369)
                                                           at com.android.internal.policy.impl.PhoneWindow.superDispatchTouchEvent(PhoneWindow.java:1719)
                                                           at android.app.Activity.dispatchTouchEvent(Activity.java:2742)
                                                           at android.support.v7.view.WindowCallbackWrapper.dispatchTouchEvent(WindowCallbackWrapper.java:60)
                                                           at android.support.v7.view.WindowCallbackWrapper.dispatchTouchEvent(WindowCallbackWrapper.java:60)
                                                           at com.android.internal.policy.impl.PhoneWindow$DecorView.dispatchTouchEvent(PhoneWindow.java:2330)
                                                           at android.view.View.dispatchPointerEvent(View.java:8666)
                                                           at android.view.ViewRootImpl$ViewPostImeInputStage.processPointerEvent(ViewRootImpl.java:4123)
                                                           at android.view.ViewRootImpl$ViewPostImeInputStage.onProcess(ViewRootImpl.java:3989)
                                                           at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:3544)
                                                           at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:3597)
                                                           at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:3563)
                                                           at android.view.ViewRootImpl$AsyncInputStage.forward(ViewRootImpl.java:3680)
                                                           at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:3571)
                                                           at android.view.ViewRootImpl$AsyncInputStage.apply(ViewRootImpl.java:3737)
                                                           at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:3544)
                                                           at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:3597)
                                                           at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:3563)
                                                           at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:3571)
                                                           at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:3544)
                                                           at android.view.ViewRootImpl.deliverInputEvent(ViewRootImpl.java:5807)
                                                           at android.view.ViewRootImpl.doProcessInputEvents(ViewRootImpl.java:5781)
                                                           at android.view.ViewRootImpl.enqueueInputEvent(ViewRootImpl.java:5752)
                                                           at android.view.ViewRootImpl$WindowInputEventReceiver.onInputEvent(ViewRootImpl.java:5897)
                                                           at android.view.InputEventReceiver.dispatchInputEvent(InputEventReceiver.java:185)
                                                           at android.os.MessageQueue.nativePollOnce(Native Method)
                                                           at android.os.MessageQueue.next(MessageQueue.java:143)
                                                           at android.os.Looper.loop(Looper.java:122)
                                                           at android.app.ActivityThread.main(ActivityThread.java:5254)
                                                           at java.lang.reflect.Method.invoke(Native Method)
                                                           at java.lang.reflect.Method.invoke(Method.java:372)
                                                           at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:903)
                                                           at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:698)
```

**Comment by ZieIony** [comment]
Thank you for the detailed bug report and your kind words! I'm glad that Carbon is useful for you.

This bug is very specific and as I can see, you're not using the most recent commit of Carbon. Could you tell me:
- manufacturer and version of Android you're using
- Maven release or GitHub commit of Carbon you're using.

As for now I think that it's the elevation system which changes the order of children using z axis translation. The fix is already in the repo. If you wish, you can check the most recent build using jitpack. Here's how to do that:

https://jitpack.io/#ZieIony/Carbon/a05d2d6934

**Comment by ZieIony** [comment]
I've released 0.12.0 which should fix this issue. As I don't have any signals from you that it's not working, I'm closing the issue.
